pkg/api/controller.go

```diff
@@ -3099,6 +3126,31 @@ func (c *Controller) ListObjects(w http.ResponseWriter, r *http.Request, reposit
 			if (params.UserMetadata == nil || *params.UserMetadata) && entry.Metadata != nil {
 				objStat.Metadata = &ObjectUserMetadata{AdditionalProperties: entry.Metadata}
 			}
+			if params.Presign != nil && *params.Presign {
+				// check if the user has read permissions for this object
+				authResponse, err := c.Auth.Authorize(ctx, &auth.AuthorizationRequest{
+					Username: user.Username,
+					RequiredPermissions: permissions.Node{
+						Permission: permissions.Permission{
+							Action:   permissions.ReadObjectAction,
+							Resource: permissions.ObjectArn(repository, entry.Path),
+						},
+					},
+				})
+				if c.handleAPIError(ctx, w, r, err) {
+					return
+				}
+				if authResponse.Allowed {
+					objStat.PhysicalAddress, err = c.BlockAdapter.GetPreSignedURL(ctx, block.ObjectPointer{
+						StorageNamespace: repo.StorageNamespace,
+						Identifier:       entry.PhysicalAddress,
+						IdentifierType:   entry.AddressType.ToIdentifierType(),
+					}, block.PreSignModeRead)
+					if c.handleAPIError(ctx, w, r, err) {
+						return
+					}
+				}
```

Comment:
When I think of it this is really neat: I will be able to read all contents of an entire directory with a single listObjects on lakeFS.